### PR TITLE
chore(dependencies): general update of dependencies

### PR DESCRIPTION
**Pull request by kikoso**
The current Pull Request introduces a general update of dependencies, and supersedes all the existing dependabot PRs.

This PR affects the `tutorials`, `snippets`, and `Api Demos` and `WearOS` projects. Changes include:

- Increased compileSdkVersion.
- Increased targetSdkVersion.
- Moves namespace from AndroidManifest to the build.gradle file.
- Increases Gradle plugin version.
- General dependencies updated.
- Updated documentation
- This supersedes all the existing dependabot PRs: #1105, #1102, #1096, #980

Note: this does not update Kotlin to the latest version (1.7.22 the 9th of December 2022), since we are waiting for the Compose compiler to add support.

### CODE REVIEW

File: ApiDemos/kotlin/app/build.gradle [hunk 1]
```diff
@@ -8,11 +8,11 @@ plugins {
 }
 
 android {
-    compileSdkVersion 31
+    compileSdkVersion 33
     defaultConfig {
         applicationId "com.example.kotlindemos"
-        minSdkVersion 19
-        targetSdkVersion 31
+        minSdkVersion 21
```
Comment (wangela) [inline ApiDemos/kotlin/app/build.gradle:14]: I don't think we're ready to bump up to minSDK 21 yet. Any reason you had to do this?

File: WearOS/README.md [hunk 2]
```diff
@@ -8,7 +8,7 @@ gradle-based Android Studio project that [supports ambient mode](https://develop
 Pre-requisites
 --------------
 
-- Android SDK v22
+- Android SDK v33
```
Comment (wangela) [inline WearOS/README.md:11]: Please verify that the sample still works with v33

Comment (kikoso) [reply WearOS/README.md]: It should be 23 (the `minSdk `on the WearOS project.

File: tutorials/java/CurrentPlaceDetailsOnMap/README.md [hunk 3]
```diff
@@ -7,7 +7,7 @@ This sample goes hand in hand with a tutorial for the Google Maps Android API:
 Prerequisites
 --------------
 
-- Android SDK v31
+- Android SDK v33
```
Comment (wangela) [inline tutorials/java/CurrentPlaceDetailsOnMap/README.md:10]: If pre-req should match minSDK, this should be 19

Comment (wangela) [reply tutorials/java/CurrentPlaceDetailsOnMap/README.md:10]: Check all READMEs for minSDK match

**Review by wangela — CHANGES_REQUESTED** [review]

File: tutorials/java/CurrentPlaceDetailsOnMap/README.md (hunk 3 above)
Comment (kikoso) [reply tutorials/java/CurrentPlaceDetailsOnMap/README.md:10]: This should match the `compileSdk `(i.e., the value specified in `compileSdkVersion `must be locally installed, and it is hence a requirement). I am reviewing it for other projects.

File: ApiDemos/kotlin/app/build.gradle (hunk 1 above)
Comment (kikoso) [reply ApiDemos/kotlin/app/build.gradle]: No. This is set back to 19.

File: WearOS/README.md (hunk 2 above)
Comment (kikoso) [reply WearOS/README.md]: Hi @wangela. We should be having here the `compileSdk `version, the `minSdk `is not a [requirement to run the samples](https://github.com/googlemaps/android-samples/pull/1107#discussion_r1046893794). I am updating this to reflect the API 31.

File: WearOS/Wearable/build.gradle [hunk 4]
```diff
@@ -21,7 +21,7 @@ plugins {
 apply plugin: 'kotlin-android'
 
 android {
-    compileSdk 31
+    compileSdk 33
```
Comment (wangela) [inline WearOS/Wearable/build.gradle:24]: For WearOS, should this be 33 or 31? targetSdk is 31 and README prerequisite says 31.

Comment (kikoso) [reply WearOS/Wearable/build.gradle:24]: This should be 33. The [guidelines for WearOS](https://developer.android.com/training/wearables/get-started/creating#wear-module) specify a different version for `targetSdk `and `compileSdk`.

Comment (wangela) [reply WearOS/Wearable/build.gradle:24]: SG. Let's add a comment above L23 so developers understand why we have different versions for `compileSdk` `targetSdk`, providing a link to https://developer.android.com/training/wearables/get-started/creating#creating as an example

**Review by wangela — APPROVED** [review]

**Review by wangela — CHANGES_REQUESTED** [review]
Build Tutorials test is [failing](https://github.com/googlemaps/android-samples/actions/runs/3735502871/jobs/6338823500). Looks like they want to move namespace definition out of AndroidManifest and into build.gradle for each of the apps under tutorials just like you did wtih snippets/app.

**Comment by kikoso** [comment]
> Build Tutorials test is [failing](https://github.com/googlemaps/android-samples/actions/runs/3735502871/jobs/6338823500). Looks like they want to move namespace definition out of AndroidManifest and into build.gradle for each of the apps under tutorials just like you did wtih snippets/app.

This has been updated. Please note that minSdk must be know 21, as a requirement for the Places SDK 3.0.0 (merged yesterday by Dependabot).

**Review by wangela — APPROVED** [review]

**Pull request merged**